### PR TITLE
[SPARK-11069] [ML] Add RegexTokenizer option to convert to lowercase

### PR DESCRIPTION
**Pull request by hhbyyh**
jira: https://issues.apache.org/jira/browse/SPARK-11069 
quotes from jira: 
Tokenizer converts strings to lowercase automatically, but RegexTokenizer does not. It would be nice to add an option to RegexTokenizer to convert to lowercase. Proposal:
call the Boolean Param "toLowercase"
set default to false (so behavior does not change)

Actually sklearn converts to lowercase before tokenizing too

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala [hunk 1]
```diff
@@ -100,10 +100,25 @@ class RegexTokenizer(override val uid: String)
   /** @group getParam */
   def getPattern: String = $(pattern)
 
-  setDefault(minTokenLength -> 1, gaps -> true, pattern -> "\\s+")
+  /**
+   * Indicates whether to convert all characters to lowercase before tokenizing.
+   * Default: false
+   * @group param
+   */
+  val toLowercase: BooleanParam = new BooleanParam(this, "toLowercase",
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala]: final val

File: mllib/src/test/scala/org/apache/spark/ml/feature/TokenizerSuite.scala [hunk 2]
```diff
@@ -69,6 +69,18 @@ class RegexTokenizerSuite extends SparkFunSuite with MLlibTestSparkContext {
     ))
     testRegexTokenizer(tokenizer2, dataset2)
   }
+
+  test("RegexTokenizer with toLowercase true"){
```
Comment (jkbradley) [inline mllib/src/test/scala/org/apache/spark/ml/feature/TokenizerSuite.scala]: style: space before brace at end of line

**Comment by jkbradley** [comment]
I'm wondering now if we should set it to convert to lowercase by default.  I know it breaks behavior, but otherwise, we'll introduce an inconsistency in the API (between Tokenizer and RegexTokenizer) which will be around for a long time.  What do you think?

**Comment by hhbyyh** [comment]
Yes, I agree.
1. Tokenizer and RegexTokenizer should have consistent behavior.
2. Whether to set toLower to true is a matter of preference. I assume for ML applications it's more common to have toLower as true.

File: mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala [hunk 3]
```diff
@@ -100,10 +100,25 @@ class RegexTokenizer(override val uid: String)
   /** @group getParam */
   def getPattern: String = $(pattern)
 
-  setDefault(minTokenLength -> 1, gaps -> true, pattern -> "\\s+")
+  /**
+   * Indicates whether to convert all characters to lowercase before tokenizing.
+   * Default: false
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala]: default needs to be updated

**Comment by jkbradley** [comment]
Looks good except that one outdated doc line

**Comment by jkbradley** [comment]
LGTM pending tests

**Comment by jkbradley** [comment]
merging with master, branch-1.6
Thank you for the PR!

**Pull request closed without merging**